recipes/nucdiff/meta.yaml

```diff
@@ -1,11 +1,11 @@
 package:
   name: nucdiff
-  version: "0.1.1"
+  version: "2.0.2"
 
 source:
-  fn: NucDiff-0.1.1.tar.gz
-  url: https://pypi.python.org/packages/c4/18/5dc9c833eba0278509b375fc69e11f19ef7a35288d623229f74730244ee8/NucDiff-0.1.1.tar.gz
-  md5: a318c1283ec1fb94ede6798588bfa804
+  fn: NucDiff-2.0.2.tar.gz
+  url: https://pypi.python.org/packages/49/bf/a83f96e9e0f5304038a2eeecfea74a10b5873dbd283d080b1a7b0e70d8fc/NucDiff-2.0.2.tar.gz
+  md5: b8c04b5353440784b82d9dce809c52ad
 
 build:
   entry_points:
```

Comment:
Please add
```yaml
  skip: True  # [not py27]
```
to the `build:` section.
Looks like the software is Python 2-only:
```
$ sed -nE 's/^\s*(\<print\>\s)/\1/p' NucDiff-2.0.2/**/*.py|sort -u
print '#-------------'
print 'BLOCKS:'
print contig_seq[b.c_st-1: min(corresp_cont_coord,b.c_end)]
print contig_seq[b.c_st-1: self.c_end]
print contig_seq[overlap_st-1: self.c_end]
print contig_seq[self.c_end+1-1: min(corresp_cont_coord,b.c_end)]
print contig_seq[self.c_end+1-1:min(corresp_cont_coord,b.c_end)]
print cont_name
print entry
print errors_list
print errors_list[-1]
print general.COMPL_STRING(ref_seq[corresp_ref_coord-1:b.r_end])
print general.COMPL_STRING(ref_seq[self.r_st-1: b.r_end])
print general.COMPL_STRING(ref_seq[self.r_st-1:b.r_end])
print general.COMPL_STRING(ref_seq[self.r_st-1:overlap_end])
print 'OUTPUT:', structure_dict[cont_name][transl_group]['output_line']
print overlap_end
print overlap_st
print 'REASON:', structure_dict[cont_name][transl_group]['reason']
print ref_seq[b.r_st-1: min(corresp_ref_coord,b.r_end)]
print ref_seq[b.r_st-1: self.r_end]
print ref_seq[overlap_st-1: self.r_end]
print ref_seq[overlap_st-1:self.r_end]
print self.errors
print self.r_end, self.c_end, b.c_st
print self.r_end, self.c_end, b.r_st
print self.r_name, b.r_st,self.r_end, cont_name, overlap_st, self.c_end
print self.r_name, overlap_st,self.r_end, cont_name, b.c_st, self.c_end
print self.r_name, self.r_st,b.r_end, cont_name, overlap_st, self.c_end
print self.r_name, self.r_st,overlap_end, cont_name, b.c_st, self.c_end
print self.r_st, self.c_end, b.c_st
print self.r_st, self.c_end, b.r_end
print '\tBLOCKS:'
print '\t', misj_group
print '\tOUTPUT:', structure_dict[cont_name][transl_group]['blocks'][misj_group]['output_line']
print transl_group
print '\tREASON:', structure_dict[cont_name][transl_group]['blocks'][misj_group]['reason']
print '\t\tBETWEEN_OUTPUT',structure_dict[cont_name][transl_group]['blocks'][misj_group]['blocks'][block_name]['between_output']
print '\t\t', block_name
print '\t\tBLOCK',structure_dict[cont_name][transl_group]['blocks'][misj_group]['blocks'][block_name]['block']
print '\t\tINVERS_OUTPUT',structure_dict[cont_name][transl_group]['blocks'][misj_group]['blocks'][block_name]['invers_output']
print '\t\tLOCAL_OUTPUT',structure_dict[cont_name][transl_group]['blocks'][misj_group]['blocks'][block_name]['local_output']
print '\t\tTRANSP_OUTPUT',structure_dict[cont_name][transl_group]['blocks'][misj_group]['blocks'][block_name]['transp_output']
```

Reply:
I ported this code to be py3 compatable with the2to3 script: https://github.com/zachcp/NucDiff

Those print staatements are within a large comment block that was presumably being used for development. I am currently using this code with python3 locally.

Reply:
> comment block

No such thing in Python. What you have there are actually strings in the middle of the code. You we're just lucky all of them are still valid; that isn't always the case, e.g.:
valid Python code:
```python
x = 1 # \xx <-- inside a comment, so everything is fine
s = r"\xx" # <-- inside a raw string, so fine again
```
invalid Python code:
```python
"""
x = 1 # \xx <-- syntax error!
s = r"\xx" # <-- syntax error!
"""
```
Always comment out code with line comments -- there just isn't another way 😉